torch/utils/cpp_extension.py

```diff
@@ -0,0 +1,60 @@
+import os.path
+import re
+import subprocess
+import warnings
+from setuptools.command.build_ext import build_ext
+
+MINIMUM_GCC_VERSION = (4, 9)
+ABI_INCOMPATIBILITY_WARNING = '''
+Your compiler ({}) may be ABI-incompatible with PyTorch.
+Please use a compiler that is ABI-compatible with GCC 4.9 and above.
+See https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html.'''
+
+
+def check_compiler_abi_compatibility(compiler):
+    '''
+    Verifies that the given compiler is ABI-compatible with PyTorch.
+
+    Arguments:
+        compiler (str): The compiler executable name to check (e.g. 'g++')
```

Comment:
i think in pretty much all build systems, `$CXX` is the first supported variable. If that's not set, look for `g++`

Reply:
oh i see you just use `compiler` which is `self.compiler.compiler_cxx[0]`, nevermind my comment